File: zmin9/BOJ_1057번_토너먼트_BruteForce.js
// 입력 처리
const fs = require('fs');
// const input = fs.readFileSync('./dev/stdin').toString().trim().split(' ').map(Number);
const input = fs.readFileSync('./input.txt').toString().trim().split(' ').map(Number);

let round = 0;

// 둘이 만나는 것은 이후 배정받을 숫자가 같은 경우 -> 승자가 가지게 되는 번호
while (input[1]!==input[2]){
    round++;
    // 경기에서 이긴 뒤 배정받을 숫자는 현재 숫자를 2로 나눈 값을 올림한 값
    input[1] = Math.ceil(input[1]/2);
    input[2] = Math.ceil(input[2]/2);
}

console.log(round);

File: jieun^,^/BOJ_1057번_토너먼트_BruteForce.py
# N명이 참가하는 토너먼트에서 (1라운드에서의) 김지민과 임한수의 번호
N, kim, lim = map(int, input().split())     # 2 <= N <= 100,000
import math
round = 1    # 김지민과 임한수가 대결하게 되는 라운드 번호 (서로 대결하지 않을 경우: -1)

kim_idx = math.ceil(kim/2)
lim_idx = math.ceil(lim/2)

while 1:
    if kim_idx == lim_idx: break

    kim_idx = math.ceil(kim_idx/2)
    lim_idx = math.ceil(lim_idx/2)

    round += 1

print(round)